### PR TITLE
Disable [QUIC] WriteCanceled_NextWriteThrows on Mock

### PR DESCRIPTION
**Pull request by karelz**
Disabled test tracked by #55995

### CODE REVIEW

File: src/libraries/System.Net.Quic/tests/FunctionalTests/QuicStreamTests.cs [hunk 1]
```diff
@@ -617,6 +617,12 @@ public async Task WritePreCanceled_Throws()
         [Fact]
         public async Task WriteCanceled_NextWriteThrows()
         {
+            // [ActiveIssue("https://github.com/dotnet/runtime/issues/55995")]
+            if (this.UseQuicImplementationProvider == QuicImplementationProviders.Mock)
```
Comment (stephentoub) [inline src/libraries/System.Net.Quic/tests/FunctionalTests/QuicStreamTests.cs:621]: Build break

File: src/libraries/System.Net.Quic/tests/FunctionalTests/QuicStreamTests.cs [hunk 2]
```diff
@@ -617,6 +617,12 @@ public async Task WritePreCanceled_Throws()
         [Fact]
         public async Task WriteCanceled_NextWriteThrows()
         {
+            // [ActiveIssue("https://github.com/dotnet/runtime/issues/55995")]
+            if (typeof(T) == typeof(MsQuicProviderFactory))
```
Comment (CarnaViire) [inline src/libraries/System.Net.Quic/tests/FunctionalTests/QuicStreamTests.cs:621]: Should it be Mock, not MsQuic?

Comment (karelz) [reply src/libraries/System.Net.Quic/tests/FunctionalTests/QuicStreamTests.cs:621]: Of course it should! Thanks for catching that!

**Review by CarnaViire — APPROVED** [review]

**Comment by karelz** [comment]
Test failures are unrelated.

**Pull request merged**